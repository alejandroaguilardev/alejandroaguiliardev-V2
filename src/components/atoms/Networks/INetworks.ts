import React from "react";

export interface INetworks {
	children: React.ReactNode;
	title: string;
    text:string;
    fab:boolean;
}
