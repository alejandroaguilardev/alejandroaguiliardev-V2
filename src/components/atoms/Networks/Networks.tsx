import { Fab, IconButton } from "@mui/material"
import { INetworks } from "./INetworks"

export const Networks = ({ children, title, fab, text }: INetworks) => {
    return (
        <>
            {fab ?
                <>
                    <Fab color="primary" size="small" aria-label={title}>
                        {children}
                    </Fab>
                </>
                :
                <>
                    <IconButton>
                        {children}
                    </IconButton>
                    {!fab && ` ${text}`}
                </>
            }
        </>
    )
}