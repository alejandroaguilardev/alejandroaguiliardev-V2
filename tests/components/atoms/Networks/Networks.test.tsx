import { render, screen } from "@testing-library/react";
import { Networks } from "../../../../src/components/atoms/Networks/Networks";

describe("Networks Test", () => {
	const title: string = "title Test";
	const text: string = "text Test";
	const fab: boolean = true;
	test("toMatchSnapshot", () => {
		const { container } = render(
			<Networks title={title} text={text} fab={fab}>
				Networks
			</Networks>
		);
		expect(container).toMatchSnapshot();
		expect(screen.getByText("Networks")).toBeTruthy();
	});

	test("Props", () => {
		render(
			<Networks title={title} text={text} fab={false}>
				Networks
			</Networks>
		);
		expect(screen.getByText(text)).toBeTruthy();
	});
});
