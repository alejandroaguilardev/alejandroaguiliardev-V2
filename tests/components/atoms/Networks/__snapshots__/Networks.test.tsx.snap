// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Networks Test toMatchSnapshot 1`] = `
<div>
  <button
    aria-label="title Test"
    class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-primary css-1dm1o0z-MuiButtonBase-root-MuiFab-root"
    tabindex="0"
    type="button"
  >
    Networks
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
